rowan/src/Rowan-GemStone-Kernel/Behavior.extension.st

```diff
@@ -101,18 +101,20 @@ Behavior >> _rwInstVar: aString constrainTo: aClass [
  will be unchanged.  Otherwise, the subclass's constraint will be changed to
  aClass."
 
-| ivInfo offset mySubclasses |
-self deprecated: 'Behavior>>instVar:constrainTo: deprecated, Constraints are no longer supported'.
+| ivInfo offset mySubclasses superConstraintClass |
 self _validatePrivilege ifFalse:[ ^ nil ].
 aClass _validateIsClass ifFalse:[ ^ nil ].
 (Symbol _existingWithAll: aString) ifNotNil:[ :aSymbol |
   ivInfo := self _ivOffsetAndConstraint: aSymbol .
 ].
 ivInfo ifNil:[ ^ self _error: #classErrNotAVar args:{ aString } ] .
 offset := ivInfo at: 1 .
-(aClass validateSubclassOf: (self superClass _namedIvConstraintAtOffset: offset)) ifFalse:[
-  ^ nil
-].
+superConstraintClass := self superClass _namedIvConstraintAtOffset: offset.
```

Comment:
Might be worth a method comment explaining why you added the Object tests. It wasn't clear why that is necessary.